kernel/packages/unity-interface/BrowserInterface.ts

```diff
@@ -377,6 +378,45 @@ export class BrowserInterface {
     globalThis.globalStore.dispatch(updateFriendship(action, userId.toLowerCase(), false))
   }
 
+  public SearchENSOwner(data: { name: string; maxResults: number }) {
+    let profilesPromise
+
+    if (/^0x[a-fA-F0-9]{40}$/.test(data.name)) {
+      profilesPromise = ProfileAsPromise(data.name, undefined, ProfileType.DEPLOYED).then((profile) => {
+        return [profile]
+      })
+    } else {
+      profilesPromise = new Promise((resolve, reject) => {
+        (async () => {
+          try {
+            const net = await getAppNetwork()
+            const owners = await fetchENSOwnersContains(ethereumConfigurations[net].names, data.name, data.maxResults)
+            const profiles: Profile[] = []
+            for (let userId of owners) {
```

Comment:
```suggestion
            await Promise.all(owners.map(userId => ProfileAsPromise(userId, undefined, ProfileType.DEPLOYED))
```
If you do this, requests execute in parallel. Otherwise, they'll go one after the other.

Reply:
changed this logic with Mr. Chamo suggestion + Promise.all